pyinteraph/reformat_interaction_network_graph.py

```diff
@@ -0,0 +1,97 @@
+import sys
+
+import argparse
+import functools
+import logging
+import networkx as nx
+import numpy as np
+import MDAnalysis as mda
+import warnings
+
+
+from pyinteraph.file_validation import ArgumentParserFileExtensionValidation
+
+warnings.filterwarnings("ignore")
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="[%(asctime)s] %(levelname)s | %(message)s",
+    datefmt="%Y/%m/%d %H:%M:%S"
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ReformatDatGraph:
+    def __init__(self, interaction_network_file, output_name, reference_structure_file=None):
+        self.reference_structure_file = reference_structure_file
+        self.interaction_network_file = interaction_network_file
+        self.output_name = output_name
+
+    @property
+    @functools.lru_cache()
+    def interaction_network(self):
+        try:
+            interaction_network = np.loadtxt(self.interaction_network_file)
+            return interaction_network
+        except:
+            sys.exit(1)
+
+    @property
+    @functools.lru_cache()
+    def reference_structure(self):
+        if not self.reference_structure_file:
+            return
+        try:
+            return mda.Universe(self.reference_structure_file)
+        except:
```

Comment:
if the user has provided a structure file and we can't find it, then we need to exit with error here. Otherwise it can be confusing if the user gets a result but it's not what they expect